### ISSUE TITLE
Euclidean Sequencer possible enhancement

### ISSUE THREAD

**Issue by ContemporaryInsanity** [countmodula/VCVRackPlugins#64]
I would like a knob that controls how the hits are distributed across the sequence, from totally Euclidean to totally random and everywhere in between.

I'd suggest that the hits change when the knob position changes as per existing knobs, an additional button to re-randomise the hits when set to non Euclidean would also be handy.

I would use this primarily to generate a random rhythm for clocking a melodic sequence though I'm sure there'd be plenty of other uses.

In summary, X randomly / pseudo randomly placed hits over Y steps, there doesn't appear to be any current module capable of doing this.

**Comment by AdamVerspaget** [comment]
Hi mate, I just want to check that I fully understand the concept. You want to be able to set the length and shift position as it currently does but randomly generate the number of hits within the given length by turning a knob? A button with an associated logic input would probably better UX for triggering the random generation of number of hits.  I don't think I could offer a knob that went from Euclidean through a blend of Euclidean and Random to Totally Random though, it would need to an all or nothing operational mode most likely by context menu selection.

**Comment by ContemporaryInsanity** [comment]
Hi Adam!

No, the number of hits is a constant set by a knob, as the number of steps is (as per current functionality), it's the distribution of the hits that is random as opposed to Euclidean.  Say you've 32 steps and 8 hits, you know you're guaranteed 8 hits across those 32 steps but can't predict where they'd be when random as opposed to Euclidean.

I could develop this myself but thought your sequencer would be a perfect match for the idea, hence the Euclidean to random knob suggestion, but a random mode would still do what I'm after, just not as interestingly.

Hope this clarifies at least a bit.

**Comment by AdamVerspaget** [comment]
Right. Got it. Thanks for the clarification.
